pkg/controller/infrastructureconfig/sync_vips.go

```diff
@@ -0,0 +1,107 @@
+	"fmt"
+	"log"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	utilslice "k8s.io/utils/strings/slices"
+)
+
+type vipsSynchronizer interface {
+	VipsSynchronize(*configv1.Infrastructure) *configv1.Infrastructure
+}
+
+type apiAndIngressVipsSynchronizer struct{}
+
+// VipsSynchronize synchronizes new API & Ingress VIPs with old fields.
+// It returns if the status was updated and the updated infrastructure object.
+//nolint:staticcheck
+func (*apiAndIngressVipsSynchronizer) VipsSynchronize(infraConfig *configv1.Infrastructure) *configv1.Infrastructure {
+	var apiVIPs, ingressVIPs *[]string // new fields
+	var apiVIP, ingressVIP *string     // old/deprecated fields
+
+	updatedInfraConfig := infraConfig.DeepCopy()
+	switch updatedInfraConfig.Status.Platform {
+	case configv1.BareMetalPlatformType:
+		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.BareMetal.APIServerInternalIPs
+		apiVIP = &updatedInfraConfig.Status.PlatformStatus.BareMetal.APIServerInternalIP
+		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.BareMetal.IngressIPs
+		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.BareMetal.IngressIP
+
+	case configv1.VSpherePlatformType:
+		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.VSphere.APIServerInternalIPs
+		apiVIP = &updatedInfraConfig.Status.PlatformStatus.VSphere.APIServerInternalIP
+		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.VSphere.IngressIPs
+		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.VSphere.IngressIP
+
+	case configv1.OpenStackPlatformType:
+		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.OpenStack.APIServerInternalIPs
+		apiVIP = &updatedInfraConfig.Status.PlatformStatus.OpenStack.APIServerInternalIP
+		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.OpenStack.IngressIPs
+		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.OpenStack.IngressIP
+
+	case configv1.OvirtPlatformType:
+		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.Ovirt.APIServerInternalIPs
+		apiVIP = &updatedInfraConfig.Status.PlatformStatus.Ovirt.APIServerInternalIP
+		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.Ovirt.IngressIPs
+		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.Ovirt.IngressIP
+
+	case configv1.NutanixPlatformType:
+		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.Nutanix.APIServerInternalIPs
+		apiVIP = &updatedInfraConfig.Status.PlatformStatus.Nutanix.APIServerInternalIP
+		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.Nutanix.IngressIPs
+		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.Nutanix.IngressIP
+
+	default:
+		// nothing to do for this platform type
+		return updatedInfraConfig
+	}
+
+	if err := syncVIPs(apiVIPs, apiVIP); err != nil {
+		log.Printf("Error on syncing api VIP fields: %v", err) // this is only a warning -> continue
+	}
+
+	if err := syncVIPs(ingressVIPs, ingressVIP); err != nil {
+		log.Printf("Error on syncing ingress VIP fields: %v", err) // this is only a warning -> continue
+	}
```

Comment:
nit: if this is a warning, I would probably choose a different wording other than `Error` for this log.

Reply:
:+1: updated it